src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -2244,3 +2247,158 @@ def lists_match(l1, l2):
         return False
 
 #endregion
+
+def _create_network_watchers(client, resource_group_name, locations, tags):
+    if resource_group_name is None:
+        raise CLIError("usage error: '--resource-group' required when enabling new regions")
+
+    from azure.mgmt.network.models import NetworkWatcher
+    for location in locations:
+        client.create_or_update(
+            resource_group_name, '{}-watcher'.format(location),
+            NetworkWatcher(location=location, tags=tags))
+
+def _update_network_watchers(client, watchers, tags):
+    from azure.mgmt.network.models import NetworkWatcher
+    for watcher in watchers:
+        id_parts = parse_resource_id(watcher.id)
+        watcher_rg = id_parts['resource_group']
+        watcher_name = id_parts['name']
+        watcher_tags = watcher.tags if tags is None else tags
+        client.create_or_update(
+            watcher_rg, watcher_name,
+            NetworkWatcher(location=watcher.location, tags=watcher_tags))
+
+def _delete_network_watchers(client, watchers):
+    for watcher in watchers:
+        from azure.cli.core.commands import LongRunningOperation
+        id_parts = parse_resource_id(watcher.id)
+        watcher_rg = id_parts['resource_group']
+        watcher_name = id_parts['name']
+        logger.warning(
+            "Disabling Network Watcher for region '%s' by deleting resource '%s'",
+            watcher.location, watcher.id)
+        LongRunningOperation()(client.delete(watcher_rg, watcher_name))
+
+def configure_network_watcher(client, locations, resource_group_name=None, enabled=None, tags=None):
+    watcher_list = list(client.list_all())
+    existing_watchers = [w for w in watcher_list if w.location in locations]
```

Comment:
You might need to ensure case insensitive comparison. Optionally, support non canonical region name such as `west us` vs `westus`, but up to you.